### ISSUE TITLE
closure_parameter_position not triggering with lack of parameter in closure

### ISSUE THREAD

**Issue by gannon-eero** [realm/SwiftLint#3225]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

What I expected from this rule was that regardless of whether there was a parameter or not (although I understand it does go against the name of the rule in a sense), it would still trigger. Instead, I would expect the rule to really apply to the `in` syntax altogether. For instance, this currently triggers:

```swift
closureFunc {
    ↓[weak self] param in
    ...
}
```

However, this does not
```swift
closureFunc {
    [weak self] in
    ...
}
```

This appears kind of odd, when you have multiple closures for a function with some with parameters and some without, such as below. The middle `onFailure` closure will trigger, but the `onSuccess` and `onReset` do not. 

```swift
thing.doStuff(onSuccess: {
    [weak self] in
    doSuccess()
}, onFailure: {
    ↓[weak self] rules in
    doFailure()
}, onReset: {
    [weak self] in
    doReset()
})
```

Will update this if I figure out a good custom rule or change that achieves this result.

**Comment by gannon-eero** [comment]
Also, in the situation where the closure is a member of a struct/class, the rule also doesn't trigger. Such as:

```swift
struct Foo {
    var completionHandler: ((Int) -> Void)? = nil
}

var f = Foo()
f.completionHandler = {
    thing in // Should trigger here
    doStuff()
}
```
